spring-integration-file/src/test/java/org/springframework/integration/file/FileWritingMessageHandlerTests.java

```diff
@@ -425,6 +429,48 @@ public void existingWritingFileNotIgnoredIfEmptySuffix() throws Exception {
 		assertFileContentIs(outFile, "foo");
 	}
 
+	@Test
+	public void noFlushAppend() throws Exception {
+		File tempFolder = this.temp.newFolder();
+		FileWritingMessageHandler handler = new FileWritingMessageHandler(tempFolder);
+		handler.setFileExistsMode(FileExistsMode.APPEND_NO_FLUSH);
+		handler.setFileNameGenerator(new FileNameGenerator() {
+
+			@Override
+			public String generateFileName(Message<?> message) {
+				return "foo.txt";
+			}
+		});
+		ThreadPoolTaskScheduler taskScheduler = new ThreadPoolTaskScheduler();
+		taskScheduler.afterPropertiesSet();
+		handler.setTaskScheduler(taskScheduler);
+		handler.setOutputChannel(new NullChannel());
+		handler.setBeanFactory(mock(BeanFactory.class));
+		handler.setFlushInterval(30000);
+		handler.afterPropertiesSet();
+		handler.start();
+		File file = new File(tempFolder, "foo.txt");
+		handler.handleMessage(new GenericMessage<String>("foo"));
+		handler.handleMessage(new GenericMessage<String>("bar"));
+		handler.handleMessage(new GenericMessage<String>("baz"));
+		handler.handleMessage(new GenericMessage<byte[]>("qux".getBytes())); // change of payload type forces flush
+		assertThat(file.length(), greaterThanOrEqualTo(9L));
+		handler.stop(); // forces flush
+		assertThat(file.length(), equalTo(12L));
+		handler.setFlushInterval(100);
+		handler.start();
+		handler.handleMessage(new GenericMessage<InputStream>(new ByteArrayInputStream("fiz".getBytes())));
+		int n = 0;
+		while (n++ < 100 && file.length() < 15) {
+			Thread.sleep(100);
+		}
+		assertThat(file.length(), equalTo(15L));
+		handler.handleMessage(new GenericMessage<InputStream>(new ByteArrayInputStream("buz".getBytes())));
+		handler.trigger(new GenericMessage<String>(file.getAbsolutePath()));
```

Comment:
`Matcher.quoteReplacement(file.getAbsolutePath())` to prevent:
```
java.util.regex.PatternSyntaxException: Illegal/unsupported escape sequence near index 3
C:\Users\abilan\AppData\Local\Temp\junit2614543298666584743\junit8807457122888397476\foo.txt
   ^

	at java.util.regex.Pattern.error(Pattern.java:1955)
```
on Windows :smile: